CMakeLists.txt

```diff
@@ -5,15 +5,37 @@ set(PACKAGE_PROJECT_ROOT_PATH
     CACHE INTERNAL "The path to the PackageProject directory"
 )
 
+# Add custom TO_INSTALL property to the project targets
+define_property(
+    TARGET
+    PROPERTY TO_INSTALL
+    BRIEF_DOCS ""
+    FULL_DOCS ""
+)
```

Comment:
Can you explain the purpose of this property?

Reply:
The idea is that one can collect the targets to install from deeply nested sub-directories automatically by setting this custom target property. This is also very convenient if some targets are optional, so the installation list is updated automatically.